### PR TITLE
Seeder and Reference Data

### PR DESCRIPTION
**Pull request by j-maynard**
This PR adds the reference data in to our code base and a way to load that seed data in to our database.  It also provides entities for us to access the reference data in our code base.

### CODE REVIEW

File: src/reference-data-entities/category.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Entity, BaseEntity, PrimaryColumn } from 'typeorm';
+
+@Entity({ name: 'categories' })
```
Comment (wheelsandcogs) [inline src/reference-data-entities/category.ts:3]: Why plural here?

Comment (j-maynard) [reply src/reference-data-entities/category.ts:3]: Its what was in the original SQL and I went with it.  Figured this was Owains area of expertise and I wouldn't meddle to much.

File: package.json [hunk 2]
```diff
@@ -11,21 +11,23 @@
     "lint:ci": "eslint .",
     "lint:fix": "eslint . --fix",
     "build": "npm-run-all clean tsc",
-    "pretest-podman": "podman compose up -d db-test",
-    "pretest": "docker compose up -d db-test",
+    "pretest": "f() { if [ -x $(command -v podman) ]; then podman compose up -d db-test; else docker compose up -d db-test; fi }; f",
```
Comment (wheelsandcogs) [inline package.json:14]: 👌

File: src/entities/dataset/user.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { BaseEntity, Column, CreateDateColumn, Entity, Index, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';
 
-@Entity({ name: 'users' })
+@Entity({ name: 'user' })
```
Comment (wheelsandcogs) [inline src/entities/dataset/user.ts:3]: 🫶

**Review by wheelsandcogs — APPROVED** [review]
LGTM - thanks for humouring my pedantry :)

**Pull request merged**